core/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -245,11 +245,22 @@ class DeltaCatalog extends DelegatingCatalogExtension
       properties: util.Map[String, String]): Table = recordFrameProfile(
         "DeltaCatalog", "createTable") {
     if (DeltaSourceUtils.isDeltaDataSourceName(getProvider(properties))) {
+
```

Comment:
can we add the comment explaining why we do this please?

Reply:
Added comment explaining why and the behaviour in both cases